shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/main/java/org/apache/shardingsphere/transaction/xa/XAShardingSphereTransactionManager.java

```diff
@@ -121,6 +122,8 @@ public void close() throws Exception {
             each.close();
         }
         cachedDataSources.clear();
-        xaTransactionManagerProvider.close();
+        if (Objects.nonNull(xaTransactionManagerProvider)) {
+            xaTransactionManagerProvider.close();
+        }
```

Comment:
`xaTransactionManagerProvider` must not be empty, maybe it is ok to close `xaTransactionManagerProvider` directly .